x/farming/keeper/plan.go

```diff
@@ -170,37 +170,35 @@ func (k Keeper) UnmarshalPlan(bz []byte) (plan types.PlanI, err error) {
 }
 
 // CreateFixedAmountPlan sets fixed amount plan.
-func (k Keeper) CreateFixedAmountPlan(ctx sdk.Context, msg *types.MsgCreateFixedAmountPlan, typ types.PlanType) (types.PlanI, error) {
+func (k Keeper) CreateFixedAmountPlan(ctx sdk.Context, msg *types.MsgCreateFixedAmountPlan, farmingPoolAcc, terminationAcc sdk.AccAddress, typ types.PlanType) (types.PlanI, error) {
 	nextId := k.GetNextPlanIdWithUpdate(ctx)
-	farmingPoolAddrAcc, err := sdk.AccAddressFromBech32(msg.FarmingPoolAddress)
-	if err != nil {
-		return nil, err
-	}
-	terminationAddrAcc := farmingPoolAddrAcc
-
-	params := k.GetParams(ctx)
-
-	balances := k.bankKeeper.GetAllBalances(ctx, farmingPoolAddrAcc)
-	diffs, hasNeg := balances.SafeSub(params.PrivatePlanCreationFee)
-	if hasNeg {
-		return nil, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFunds, "lack of %s coins to pay private plan createion fee", diffs.String())
-	}
+	// TODO: if public proposal, It should be choose custom terminationAddr
+
+	// TODO: check if private, not equal msg.creator, terminationAcc
```

Comment:
Is this comment for whether or not we should add validation logic for `msg.creator` is equal to `terminationAcc`?

Reply:
Yes, I'll add validation logic here